README.md

```diff
@@ -8,10 +8,119 @@ Ballerina Constraint Library
   [![GitHub issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/constraint.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fconstraint)
 
-This library provides features to validate the values that have been assigned to Ballerina types.
-
 The Ballerina `constraint` library facilitates APIs to do validations on the Ballerina types further with the use of annotations.
 
+### Constraint annotations
+
+This library provides the following annotations to validate the values that have been assigned to Ballerina types.
+
+| Ballerina Type                    | Annotation           | Supported Constraints                                                                                                  |
+|-----------------------------------|----------------------|------------------------------------------------------------------------------------------------------------------------|
+| `int`                             | `@constraint:Int`    | `minValue`, `maxValue`, `minValueExclusive`, `maxValueExclusive`, `maxDigits`, `maxIntegerDigits`, `maxFractionDigits` |
+| `float`                           | `@constraint:Float`  | `minValue`, `maxValue`, `minValueExclusive`, `maxValueExclusive`, `maxDigits`, `maxIntegerDigits`, `maxFractionDigits` |
+| `int`&#124;`float`&#124;`decimal` | `@constraint:Number` | `minValue`, `maxValue`, `minValueExclusive`, `maxValueExclusive`, `maxDigits`, `maxIntegerDigits`, `maxFractionDigits` |
+| `string`                          | `@constraint:String` | `length`, `minLength`, `maxLength`, `pattern`                                                                          |
+| `any[]`                           | `@constraint:Array`  | `length`, `minLength`, `maxLength`                                                                                     |
+| `constraint:Date`                 | `@constraint:Date`   | `option` - `PAST` or `FUTURE` or `PAST_OR_PRESENT` or `FUTURE_OR_PRESENT`                                              |
+
+The following example demonstrates how to apply constraint annotations to Ballerina types.
+
+```ballerina
+@constraint:String { pattern: re `^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$` }
+public type Email string;
+
+@constraint:Date { option: constraint:PAST }
+public type DOB time:Date;
+
+public type Person record {|
+    @constraint:String { minLength: 5, maxLength: 10 }
+    string name;
+    @constraint:Int { minValue: 18, maxValue: 60 }
+    int age;
+    Email email;
+    @constraint:Array { minLength: 1, maxLength: 5 }
+    string[] phoneNumbers;
+    DOB dob;
+|};
+```
+
+### Constraint validation
+
+The `validate` function in this library can be used to validate the values that have been assigned to Ballerina types with respect to the constraints defined using the annotations.
```

Comment:
Would change similarly. Since we aren't assigning values to type, but creating values of those types or providing values belonging to those types.

Reply:
Fixed via [d30a43e](https://github.com/ballerina-platform/module-ballerina-constraint/pull/123/commits/d30a43eca8463f89ed28fd35c2f6ee3d84b0b4c2)